datasets/common_gen/common_gen.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import, division, print_function
+
+import json
+import os
+import random
+
+import nlp
+
+random.seed(42)  # This is important, to ensure the same order for concept sets as the official script.
+
+_CITATION = """\
+@article{lin2019comgen,
+     author = {Bill Yuchen Lin and Ming Shen and Wangchunshu Zhou and Pei Zhou and Chandra Bhagavatula and Yejin Choi and Xiang Ren},
+     title = {CommonGen: A Constrained Text Generation Challenge for Generative Commonsense Reasoning},
+     journal = {CoRR},
+     volume = {abs/1911.03705},
+     year = {2019}
+}
+"""
+
+_DESCRIPTION = """\
+CommonGen is a constrained text generation task, associated with a benchmark dataset, 
+to explicitly test machines for the ability of generative commonsense reasoning. Given 
+a set of common concepts; the task is to generate a coherent sentence describing an 
+everyday scenario using these concepts.
+
+CommonGen is challenging because it inherently requires 1) relational reasoning using 
+background commonsense knowledge, and 2) compositional generalization ability to work 
+on unseen concept combinations. Our dataset, constructed through a combination of 
+crowd-sourcing from AMT and existing caption corpora, consists of 30k concept-sets and 
+50k sentences in total.
+"""
```

Comment:
Feel free to edit this if you think it's necessary. @yuchenlin @MichaelZhouwang

Reply:
Thank you for the help!